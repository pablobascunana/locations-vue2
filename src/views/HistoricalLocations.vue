<template>
  <div>
    <LocationsTable
      :markers="markers"
      :userUuid="userUuid"
      @updateMarkers="removeMarkersHandler"
    ></LocationsTable>
    <AddLocationDialog
      :markers="markers"
      :userUuid="userUuid"
      @updateMarkers="addMarkersHandler"
    ></AddLocationDialog>
  </div>
</template>

<script>
import Api from '@/api/services';
import LocationsTable from '@/components/LocationsTable'
import { mapState } from 'vuex'
export default {
  name: 'locationsHistory',
  components: {
    AddLocationDialog: () => import('@/components/dialogs/AddLocation'),
    LocationsTable
  },
  data() {
    return {
      markers: []
    }
  },
  computed: {
    ...mapState('user', ['userUuid']),
  },
  mounted() {
    this.getLocations();
  },
  methods: {
    async getLocations() {
      try {
        let { data } = await Api.getLocations(this.userUuid);
        this.markers = data.markers;
      } catch (error) {
        this.showErrorSnackBar(error.response.status);
      }
    },
    showErrorSnackBar(status) {
      if (status !== 404) {
        this.EventBus.$emit('showSnackBar', { 
          snackBarMessage: this.$t('locations.error.getGeneric'),
          snackBarColor: 'error'
        });
      }
    },
    removeMarkersHandler(markers) {
      this.updateMarkers(markers);
    },
    addMarkersHandler(markers) {
      this.updateMarkers(markers);
    },
    updateMarkers(markers) {
      this.markers = markers;
    }
  }
}
</script>

<style>

</style>