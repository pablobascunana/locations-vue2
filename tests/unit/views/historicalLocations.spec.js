import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

import user from '@/store/modules/user.js'

import HistoricalLocations from '@/views/HistoricalLocations.vue'

import * as historicalLocationsMock from '../mocks/historicalLocations.mock.js'

const Plugins = {
    install(Vue) {
      Vue.prototype.EventBus = new Vue();
    }
};

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Plugins);
let vuetify = new Vuetify();

describe('Historical locations', () => {
    let wrapper;
    let store;
    let state = {};
    let actions = {};
    let modules = {
        user
    };
    const mocks = {
        EventBus: {
            $emit: jest.fn(),
        },
        $t: () => {'some specific text in any language'}
    };
    beforeEach(() => {
        store = new Vuex.Store({ state, actions, modules });

        wrapper = shallowMount(HistoricalLocations, {
            localVue,
            vuetify,
            store,
            mocks
        });
    });

    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });

    it('Call to show error snack bar function', () => {
        const showErrorSnackBarFn = jest.spyOn(wrapper.vm, 'showErrorSnackBar');
        wrapper.vm.showErrorSnackBar(500);
        expect(showErrorSnackBarFn).toBeCalled();
    });

    it('Call to update markers handler function', () => {
        const updateMarkersHandlerFn = jest.spyOn(wrapper.vm, 'updateMarkersHandler');
        wrapper.vm.updateMarkersHandler(historicalLocationsMock.MARKER);
        expect(updateMarkersHandlerFn).toBeCalled();
        expect(wrapper.vm.markers).toStrictEqual(historicalLocationsMock.MARKER)
    });
});