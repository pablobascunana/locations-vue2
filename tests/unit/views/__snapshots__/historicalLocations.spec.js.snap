// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Historical locations Should match snapshot 1`] = `
<div>
  <locationstable-stub markers=""></locationstable-stub>
  <v-dialog max-width="90%">
    <v-card>
      <v-card-title>
        <v-spacer></v-spacer>
        <v-icon id="close" aria-label="Close">mdi-close</v-icon>
      </v-card-title>
      <v-card-text class="pt-0 pb-0">
        <locationsmap-stub markers=""></locationsmap-stub>
      </v-card-text>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn id="exit" color="primary" text="">
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</div>
`;
